### PR TITLE
feat: 6 add pagination and swagger

### PR DESCRIPTION
**Pull request by EduarfF**

### CODE REVIEW

File: src/main/java/org/example/springbootintro/controller/BookController.java [hunk 1]
```diff
@@ -67,6 +76,7 @@ public BookDto updateById(@PathVariable Long id,
 
     @GetMapping("/search")
     @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Search a book", description = "Search a book by author or title or both")
     public List<BookDto> search(@Valid BookSearchParametersDto bookSearchParametersDto) {
```
Comment (andrii-hoienko) [inline src/main/java/org/example/springbootintro/controller/BookController.java]: Add pageable here as well

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments

File: src/main/java/org/example/springbootintro/controller/BookController.java [hunk 2]
```diff
@@ -67,7 +76,11 @@ public BookDto updateById(@PathVariable Long id,
 
     @GetMapping("/search")
     @ResponseStatus(HttpStatus.OK)
-    public List<BookDto> search(@Valid BookSearchParametersDto bookSearchParametersDto) {
-        return bookService.search(bookSearchParametersDto);
+    @Operation(summary = "Search books", description = "Search books by author or title or both")
+    public List<BookDto> search(
+            Pageable pageable,
+            @Valid BookSearchParametersDto bookSearchParametersDto
```
Comment (karma-o) [inline src/main/java/org/example/springbootintro/controller/BookController.java:82]: having a search parameters documentation is usually very useful for the whole team

File: src/main/java/org/example/springbootintro/controller/BookController.java [hunk 3]
```diff
@@ -35,30 +39,35 @@ public void initBinder(WebDataBinder binder) {
...
+    @Operation(summary = "Delete book by id", description = "Delete book by id")
     public void deleteById(@PathVariable Long id) {
         bookService.deleteById(id);
     }
 
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Update book by id", description = "Update book by id")
     public BookDto updateById(@PathVariable Long id,
                               @RequestBody @Valid CreateBookRequestDto bookDto
```
Comment (karma-o) [inline src/main/java/org/example/springbootintro/controller/BookController.java:72]: The same goes for the request DTOs, we can also add some description for some of those fields as well, like defining what formats of ISBN we accept for example

**Review by karma-o — APPROVED** [review]
Approved with a couple more suggestions from me

**Pull request merged**